spec/code_block_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+require 'kramdown'
+require 'ap'
+require 'pp'
+site = File.join(File.dirname(__FILE__), '..','content', '**', '*.md')
+PAGES = Dir.glob(site)
+
+def extract_code(element,prev_element)
+  codes = [] 
+  if element.type == :codespan \
+    && !element.value.nil? \
```

Comment:
Does this ever occur? It could probable be removed and `.value` converted to empty string to return failure as the `:codespan` is empty.

_BTW. I quite prefer `codespan` instead of `codeblock`._